src/client/components/MealsWithId.js

```diff
@@ -0,0 +1,152 @@
+
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Link } from 'react-router-dom';
+import "./home.css";
+
+export const MealsWithId = () => {
+    const { id } = useParams();
+    const [availableReservation, setAvailableReservation] = useState([]);
+
+    const [title, setNewTitle] = useState([]);
```

Comment:
Same here not a very fitting name. And as it is only 1 meal you are fetching consider that it should not be an array but rather an object
```suggestion
    const [meal, setMeal] = useState([]);
```